Update sorted-sets.md to use bike racing story and hugo short codes (#2541)
redis/redis-doc: docs/data-types/sorted-sets.md

* If B and A are two elements with a different score, then A > B if A.score is > B.score.
* If B and A have exactly the same score, then A > B if the A string is lexicographically greater than the B string. B and A strings can't be equal since sorted sets only have unique elements.

Let's start with a simple example, we'll add all our racers and the score they got in the first race:

{{< clients-example ss_tutorial zadd >}}
> ZADD racer_scores 10 "Norem"
(integer) 1
> ZADD racer_scores 12 "Castilla"
(integer) 1
> ZADD racer_scores 8 "Sam-Bodden" 10 "Royce" 6 "Ford" 14 "Prickett"
(integer) 4

> {{ /clients-example> }}

Reviewer: The tags needs to be fully enclosed in < >. Looks like the there's the same issue all the others below.

```suggestion
{{ < /clients-example > }}
```

dual-ported data structure containing both a skip list and a hash table, so
every time we add an element Redis performs an O(log(N)) operation. That's
good, but when we ask for sorted elements Redis does not have to do any work at

> all, it's already all sorted. Note that the `ZRANGE` order is low to high, while the `ZREVRANGE` order is high to low:

Reviewer: ```suggestion
all, it's already sorted. Note that the `ZRANGE` order is low to high, while the `ZREVRANGE` order is high to low:
```

> ZRANGE racer_scores 0 -1
1) "Ford"
2) "Sam-Bodden"
3) "Norem"
4) "Royce"
5) "Castilla"
6) "Prickett"
> ZREVRANGE racer_scores 0 -1
1) "Prickett"
2) "Castilla"
3) "Royce"
4) "Norem"
5) "Sam-Bodden"
6) "Ford"
{{ /clients-example> }}

Note: 0 and -1 means from element index 0 to the last element (-1 works
here just as it does in the case of the `LRANGE` command).

It is possible to return scores as well, using the `WITHSCORES` argument:

{{< clients-example ss_tutorial zrange_withscores >}}
> ZRANGE racer_scores 0 -1 withscores
 1) "Ford"
 2) "6"
 3) "Sam-Bodden"
 4) "8"
 5) "Norem"
 6) "10"
 7) "Royce"
 8) "10"
 9) "Castilla"
10) "12"
11) "Prickett"
12) "14"
{{ /clients-example> }}

### Operating on ranges

Sorted sets are more powerful than this. They can operate on ranges.
Let's get all the racers with 10 or fewer points. We
use the `ZRANGEBYSCORE` command to do it:

{{< clients-example ss_tutorial zrangebyscore >}}
> ZRANGEBYSCORE racer_scores -inf 10
1) "Ford"
2) "Sam-Bodden"
3) "Norem"
4) "Royce"
{{ /clients-example> }}

We asked Redis to return all the elements with a score between negative
infinity and 10 (both extremes are included).

> To remove an element we'd simply call `ZREM` with the racers name. 

Reviewer: ```suggestion
To remove an element we'd simply call `ZREM` with the racer's name. 
```

It's also possible to remove ranges of elements. Let's remove racer Castilla along with all
the racers with strictly fewer than 10 points:

{{< clients-example ss_tutorial zremrangebyscore >}}
> ZREM racer_scores "Castilla"
(integer) 1
> ZREMRANGEBYSCORE racer_scores -inf 9
(integer) 2
> ZRANGE racer_scores 0 -1
1) "Norem"
2) "Royce"
3) "Prickett"
{{ /clients-example> }}

`ZREMRANGEBYSCORE` is perhaps not the best command name,
but it can be very useful, and returns the number of removed elements.

Another extremely useful operation defined for sorted set elements
is the get-rank operation. It is possible to ask what is the

> position of an element in the set of the ordered elements. 

Reviewer: ```suggestion
position of an element in the set of ordered elements. 
```

`ZREVRANGEBYLEX`, `ZREMRANGEBYLEX` and `ZLEXCOUNT`.

For example, let's add again our list of famous hackers, but this time

> use a score of zero for all the elements. We'll see that because of the sorted sets ordering rules, they are already sorted lexicographically. Using `ZRANGEBYLEX` we can ask for lexicographical ranges:

Reviewer: ```suggestion
using a score of zero for all the elements. We'll see that because of the sorted set's ordering rules, they are already sorted lexicographically. Using `ZRANGEBYLEX` we can ask for lexicographical ranges:
```

## Examples

> * There are two ways we can use a sorted set to represent a leaderbaord. If we know a racers new score, we can update it directly via the `ZADD` command. However if we want to add points to an existing score, we can use the `ZINCRBY` command.

Reviewer: ```suggestion
* There are two ways we can use a sorted set to represent a leaderbaord. If we know a racer's new score, we can update it directly via the `ZADD` command. However, if we want to add points to an existing score, we can use the `ZINCRBY` command.
```

{{< clients-example ss_tutorial leaderboard >}}
> ZADD racer_scores 100 "Wood"
(integer) 1
> ZADD racer_scores 100 "Henshaw"
(integer) 1
> ZADD racer_scores 150 "Henshaw"
(integer) 0
> ZINCRBY racer_scores 50 "Wood"
"150"
> ZINCRBY racer_scores 50 "Henshaw"
"200"
{{ /clients-example> }}

> You'll see that `ZADD` returns 0 when the member already exists, and the score is updated while `ZINCRBY` returns the new score. The score for racer Henshaw went from 100, was changed to 150 with no regard for what score was there before, and then was incremented by 50 to 200.

Reviewer: ```suggestion
You'll see that `ZADD` returns 0 when the member already exists (the score is updated), while `ZINCRBY` returns the new score. The score for racer Henshaw went from 100 to 150 with no regard for what score was there before, and then was incremented by 50 to 200.
```
